lib/data/services/remote/api_service.dart

```diff
@@ -0,0 +1,175 @@
+import 'package:appetizer/data/constants/api_endpoints.dart';
+import 'package:appetizer/domain/models/appetizer_version.dart';
+import 'package:appetizer/domain/models/feedback/appetizer_feedback.dart';
+import 'package:appetizer/domain/models/feedback/feedback_response.dart';
+import 'package:appetizer/domain/models/leaves/paginated_leaves.dart';
+import 'package:appetizer/domain/models/menu/week_menu.dart';
+import 'package:appetizer/domain/models/transaction/faq.dart';
+import 'package:appetizer/domain/models/transaction/paginated_yearly_rebate.dart';
+import 'package:appetizer/domain/models/user/notification.dart';
+import 'package:appetizer/domain/models/user/oauth_user.dart';
+import 'package:appetizer/domain/models/user/user.dart';
+import 'package:dio/dio.dart' hide Headers;
+import 'package:retrofit/retrofit.dart';
+import 'dart:async';
+
+part 'api_service.g.dart';
+
+@RestApi(baseUrl: ApiEndpoints.baseUrl)
+abstract class ApiService {
+  factory ApiService(Dio dio, {String baseUrl}) = _ApiService;
+
+  @GET(ApiEndpoints.checkVersion)
+  Future<AppetizerVersion> checkVersion(
+    @Path("platform") String platform,
+    @Path("versionNumber") String versionNumber,
+  );
+
+  // TODO: add token to all headers
```

Comment:
@NanoNish you don't have to do this. I have already implemented in the dio interceptors.